sqlmesh/utils/__init__.py

```diff
@@ -152,3 +152,28 @@ def double_escape(s: str) -> str:
 
 def nullsafe_join(join_char: str, *args: t.Optional[str]) -> str:
     return join_char.join(filter(None, args))
+
+
+# SO: https://stackoverflow.com/questions/7204805/how-to-merge-dictionaries-of-dictionaries
+def merge_dicts(dict1: t.Dict[str, t.Any], dict2: t.Dict[str, t.Any]) -> t.Dict[str, t.Any]:
+    """Recursively merge two dictionaries."""
+
+    def _merge_dicts(
+        dict1: t.Dict[str, t.Any], dict2: t.Dict[str, t.Any]
+    ) -> t.Generator[t.Tuple[str, t.Any], None, None]:
+        for k in set(dict1.keys()).union(dict2.keys()):
```

Comment:
don't need keys

Reply:
or even better just do

for k in (dict1.keys() | dict2.keys())

Reply:
You should double check your supported python version range for sqlmesh before using the dict union |
It was introduced in maybe 3.9